src/xhr.js

```diff
@@ -144,6 +152,28 @@ class Xhr {
   }
 
   /**
+   * Fetches and returns the `FetchResponse` object.
+   *
+   * See https://developer.mozilla.org/en-US/docs/Web/API/GlobalFetch/fetch
+   *
+   * See `fetchAmpCors_` for more detail.
+   *
+   * @param {string} input
+   * @param {?FetchInitDef=} opt_init
+   * @return {!Promise<!FetchResponse>}
+   */
+  fetchHtml(input, opt_init) {
+    const init = opt_init || {};
+    init.responseType = 'document';
+    init.method = normalizeMethod_(init.method);
+    init.headers = init.headers || {};
+    init.headers['Accept'] = 'text/html';
+    init.headers['Content-Type'] = 'text/html;charset=utf-8';
+
+    return this.fetchAmpCors_(input, init);
```

Comment:
@dvoytenko so i need the original xhr to make decisions based on the status code.